### PR TITLE
refactor: LT-998 Merge draft model into application

### PR DESCRIPTION
**Pull request by andytaylorkainos**

### CODE REVIEW

File: apply_for_a_licence/views/common.py [hunk 1]
```diff
@@ -51,36 +52,44 @@ def get_licence_overview(request, kwargs, errors=None):
...
-        end_user_document, _ = get_end_user_document(request, draft_id)
-        end_user_document = end_user_document.get('document')
-    else:
-        end_user_document = None
-    consignee = draft.get('consignee')
-    if consignee:
-        consignee_document, _ = get_consignee_document(request, draft_id)
-        consignee_document = consignee_document.get('document')
+
+    if draft['licence_type']['key'] == STANDARD_LICENCE:
```
Comment (craiga) [inline apply_for_a_licence/views/common.py]: Above we call `draft = data.get('draft')`, so `draft` could theoretically be `None` here. Is this possible? Should there be a `if draft is None` check?

Comment (andytaylorkainos) [reply apply_for_a_licence/views/common.py]: Redirecting back to home-page if the draft isn't found for now :)

File: apply_for_a_licence/views/common.py [hunk 2]
```diff
@@ -51,36 +52,44 @@ def get_licence_overview(request, kwargs, errors=None):
...
+        if consignee:
+            consignee_document, _ = get_consignee_document(request, draft_id)
+            consignee_document = consignee_document.get('document')
+
+        for good in goods['goods']:
+            if not good['good']['is_good_end_product']:
+                ultimate_end_users_required = True
     else:
-        consignee_document = None
+        goodstypes, _ = get_draft_goods_type(request, draft_id)
```
[the remark this answers is not in the document]
Comment (andytaylorkainos) [reply apply_for_a_licence/views/common.py]: Done!

File: apply_for_a_licence/views/common.py [hunk 3]
```diff
@@ -48,39 +49,52 @@ def check_all_parties_have_a_document(parties):
 
 def get_licence_overview(request, kwargs, errors=None):
     draft_id = str(kwargs['pk'])
-    data, _ = get_draft(request, draft_id)
+    data, status_code = get_draft(request, draft_id)
+
+    if status_code is not 200:
```
Comment (BenAndrew-Kainos) [inline apply_for_a_licence/views/common.py]: Please use !=

File: apply_for_a_licence/views/common.py [hunk 4]
```diff
@@ -48,39 +49,52 @@ def check_all_parties_have_a_document(parties):
...
-    goods, _ = get_draft_goods(request, draft_id)
-    ultimate_end_users_required = False
-    countries, _ = get_draft_countries(request, draft_id)
-    goodstypes, _ = get_draft_goods_type(request, draft_id)
     external_locations, _ = get_external_locations_on_draft(request, draft_id)
-    ultimate_end_users, _ = get_ultimate_end_users(request, draft_id)
-    third_parties, _ = get_third_parties(request, draft_id)
     additional_documents, _ = get_additional_documents(request, draft_id)
 
+    countries = {'countries': list()}
```
Comment (BenAndrew-Kainos) [inline apply_for_a_licence/views/common.py]: list() instead of []... very interesting

File: apply_for_a_licence/views/common.py [hunk 5]
```diff
@@ -113,7 +129,7 @@ def post(self, request, **kwargs):
         draft_id = str(kwargs['pk'])
         data, status_code = submit_draft(request, draft_id)
 
-        if status_code is not 201:
+        if status_code is not 200:
```
Comment (BenAndrew-Kainos) [inline apply_for_a_licence/views/common.py]: !=

File: ui_automation_tests/helpers/seed_data.py [hunk 6]
```diff
@@ -364,8 +364,8 @@ def add_draft(self, draft=None, good=None, enduser=None, ultimate_end_user=None,
     def submit_application(self, draft_id=None):
         self.log('submitting application: ...')
         draft_id_to_submit = draft_id if None else self.context['draft_id']  # noqa
-        data = {'id': draft_id_to_submit}
-        response = self.make_request('POST', url='/applications/', headers=self.export_headers, body=data)
+        response = self.make_request('PUT', url='/applications/' + draft_id_to_submit + '/submit/',
+                                     headers=self.export_headers)
```
Comment (BenAndrew-Kainos) [inline ui_automation_tests/helpers/seed_data.py]: Will add these changes to submodule

File: apply_for_a_licence/views/common.py (hunk 3 above)
Comment (andytaylorkainos) [reply apply_for_a_licence/views/common.py]: done

File: apply_for_a_licence/views/common.py (hunk 4 above)
Comment (andytaylorkainos) [reply apply_for_a_licence/views/common.py]: done

File: apply_for_a_licence/views/common.py (hunk 5 above)
Comment (andytaylorkainos) [reply apply_for_a_licence/views/common.py]: done

File: ui_automation_tests/helpers/seed_data.py (hunk 6 above)
Comment (andytaylorkainos) [reply ui_automation_tests/helpers/seed_data.py]: 👍

**Pull request merged**